### PR TITLE
forensic haven cleanup

### PR DESCRIPTION
**Pull request by jerryblakley**

### CODE REVIEW

File: src/filecache.c [hunk 1]
```diff
@@ -1504,19 +1578,38 @@ void filecache_forensic_haven(const char *cache_path, filecache_t *cache, const
...
+                    "%s: error on clear_files: %s -- %d: %s", 
+                    fname, subgerr->message, subgerr->code, g_strerror(subgerr->code));
+            break;
+        }
+        hours /= 4;
+        log_print(LOG_NOTICE, SECTION_FILECACHE_FILE, 
+                "%s: files_left: %d; files_kept: %d; hours: %d", 
+                fname, files_left, files_kept, hours);
+    }
+    free(newpath);
```
Comment (kibra) [inline src/filecache.c]: How is this not a double free?

File: src/filecache.c [hunk 2]
```diff
@@ -1504,19 +1578,38 @@ void filecache_forensic_haven(const char *cache_path, filecache_t *cache, const
...
 
 finish:
     if (fd >= 0) close(fd);
     free(buf);
     free(newpath);
+    newpath = NULL;
     free(pdata);
-    log_print(LOG_DEBUG, SECTION_FILECACHE_CACHE, "filecache_forensic_haven: exiting for %s", path);
+    // Cleanup older entries in the forensic haven
+    asprintf(&newpath, "%s/%s/", cache_path, forensic_haven_dir);
```
Comment (kibra) [inline src/filecache.c]: Don'e we need to realloc newpath?

Comment (jerryblakley) [inline src/filecache.c]: If I'm not mistaken, asprintf does the alloc

File: src/filecache.c (hunk 1 above)
Comment (jerryblakley) [inline src/filecache.c]: We free a previous alloc. Then we realloc via asprintf, then we free again. At least that's the theory.

File: src/filecache.c (hunk 2 above)
Comment (kibra) [inline src/filecache.c]: Yup, I'm forgetting my c stdlib :) or is that stdio?

**Comment by kibra** [comment]
+1

:lock:

**Pull request merged**